packages/icrc-ledger-agent/src/lib.rs

```diff
@@ -341,42 +341,62 @@ impl Icrc1Agent {
         };
         self.verify_root_hash(&certificate, &hash_tree.digest())
             .await?;
+        let last_block_index_encoded = match lookup_leaf(&hash_tree, "last_block_index")? {
+            Some(last_block_index) => last_block_index,
+            None => {
+                return Ok(None);
+            }
+        };
 
-        let last_block_hash_vec = lookup_leaf(&hash_tree, "tip_hash")?;
-        if let Some(last_block_hash_vec) = last_block_hash_vec {
-            let last_block_hash: Hash = match last_block_hash_vec.clone().try_into() {
-                Ok(last_block_hash) => last_block_hash,
+        fn convert_block_hash(block_hash: Vec<u8>) -> Result<Hash, Icrc1AgentError> {
+            match block_hash.clone().try_into() {
+                Ok(last_block_hash) => Ok(last_block_hash),
                 Err(_) => {
-                    return Err(Icrc1AgentError::VerificationFailed(format!(
-                "DataCertificate last_block_hash bytes: {}, cannot be decoded as last_block_hash",
-                hex::encode(last_block_hash_vec)
-            )))
+                    Err(Icrc1AgentError::VerificationFailed(format!(
+                    "DataCertificate last_block_hash bytes: {}, cannot be decoded as last_block_hash",
+                    hex::encode(block_hash)
+                )))
```

Comment:
I believe this could be simplified as 
`block_hash.clone().try_into().or(Err(Icrc1AgentError::...`